user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -656,8 +720,14 @@ define([
 
     fetchData: function (path, callback) {
       var myself = this,
-          url = this.getFileListRequest(FileBrowser.encodePathComponents(path == null ? ":" : Encoder.encodeRepositoryPath(path))),
+          url = this.getFileListRequest(path == null ? ":" : encodeGenericPath(path)),
```

Comment:
Please, see other related comment about not using `:` explicitly.

Reply:
so, this is effectively saying that if the path is null, fetch the root of the tree... I think it's fair to keep this?

This has a much longer history, it goes back to [the creation](https://github.com/pentaho/pentaho-platform/blame/f27826561d8123dfc19fbacb25375e963be02036/user-console/source/org/pentaho/mantle/public/browser/js/browser.js) of the current Browse perspective.

I'm not saying we _shouldn't_ question things just because they've been around for a long time, but I am extra cautious to change something like this. `path` variable is [initialized](https://github.com/pentaho/pentaho-platform/blob/7e7c38f50606e845c5bb5c717cd90319191d6ce9/user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js#L557) to "/", and it is _probably_ never going to be null? I just don't have the same level of confidence to make this change right here right now. I have already changed so much, and while I do want to "sneak in" some quality of life/feature improvements as I can, I think I will leave this one as-is.

Reply:
Ok, that's reasonable. Makes sense as a default. I thought one of these instances of this default was only used to get sub-tree, for which a value would always need to be specified. If this one is (also) used to get the root tree, then, yes.